packages/grid/x-data-grid/src/hooks/features/columns/useGridColumnGrouping.ts

```diff
@@ -0,0 +1,177 @@
+import {
+  GridColumnGroupingModel,
+  GridColumnNode,
+  isLeaf,
+} from '../../../models/gridColumnGrouping';
+import { gridColumnGroupsLookupSelector, gridColumnLookupSelector } from './gridColumnsSelector';
+import { GridColumnGroupLookup } from './gridColumnsInterfaces';
+import { GridColumnGroupingApi } from '../../../models/api/gridColumnGroupingApi';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridStateColDef, GridColDef } from '../../../models/colDef';
+
+export function hasGroupPath(
+  lookupElement: GridColDef | GridStateColDef,
+): lookupElement is GridStateColDef {
+  return (<GridStateColDef>lookupElement).groupPath !== undefined;
+}
+
+type UnwrappedGroupingModel = { [key: string]: string[] };
+
+const recurrentUnwrapGroupingColumnModel = (
+  columnGroupNode: GridColumnNode,
+  parents: any,
+): UnwrappedGroupingModel => {
+  if (isLeaf(columnGroupNode)) {
+    return { [columnGroupNode.field]: parents };
+  }
+  const rep: UnwrappedGroupingModel = {};
+  const { groupId, children } = columnGroupNode;
+  children.forEach((child) => {
+    const unwrappedSubTree = recurrentUnwrapGroupingColumnModel(child, [...parents, groupId]);
+    Object.entries(unwrappedSubTree).forEach(([key, value]) => {
+      if (rep[key] !== undefined) {
+        throw new Error(
+          [
+            `MUI DataGrid - column grouping: duplicated field`,
+            `column field ${key} occurres two times in the grouping model:`,
+            `- ${rep[key].join(' > ')}`,
+            `- ${value.join(' > ')}`,
+          ].join('\n'),
+        );
+      }
+      rep[key] = value;
+    });
+  });
+  return rep;
+};
+
+export const unwrapGroupingColumnModel = (
+  columnGroupingModel?: GridColumnGroupingModel,
+): UnwrappedGroupingModel => {
+  if (!columnGroupingModel) {
+    return {};
+  }
+
+  const rep: UnwrappedGroupingModel = {};
+
+  columnGroupingModel.forEach((columnGroupNode) => {
+    const unwrappedSubTree = recurrentUnwrapGroupingColumnModel(columnGroupNode, []);
+    Object.entries(unwrappedSubTree).forEach(([key, value]) => {
+      if (rep[key] !== undefined) {
+        throw new Error(
+          [
+            `MUI: column grouping has duplicated field.`,
+            `column field ${key} occurres two times in the grouping model:`,
+            `- ${rep[key].join(' > ')}`,
+            `- ${value.join(' > ')}`,
+          ].join('\n'),
+        );
+      }
+      rep[key] = value;
+    });
+  });
+
+  return rep;
+};
+
+const createGroupLookup = (columnGroupingModel: GridColumnNode[]): GridColumnGroupLookup => {
+  let groupLookup: GridColumnGroupLookup = {};
+
+  columnGroupingModel.forEach((node) => {
+    if (isLeaf(node)) {
+      return;
+    }
+    const { groupId, children, ...other } = node;
+    if (!groupId) {
+      throw new Error(
+        'MUI-DataGrid: an element of the columnGroupingModel does not have either `field` or `groupId`',
```

Comment:
```suggestion
        'MUI: An element of the columnGroupingModel does not have either `field` or `groupId`',
```

If we want to change the prefix we should do it everwhere (in this PR you use `MUI`, `MUI DataGrid` and `MUI-DataGrid`)